src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2568 @@
+
+        elif isinstance(module, MaskFormerFPNConvLayer):
+            nn.init.xavier_uniform_(module[0].weight, gain=xavier_std)
+        # The MLP head
+        elif isinstance(module, MaskformerMLPPredictionHead):
+            # I was not able to find the correct initializer in the original implementation
+            # we'll use xavier
+            for layer in module:
+                nn.init.xavier_uniform_(layer[0].weight, gain=xavier_std)
+                nn.init.constant_(layer[0].bias, 0)
+
+
+@add_start_docstrings(
+    "The bare MaskFormer Model outputting raw hidden-states without any specific head on top.",
+    MASKFORMER_START_DOCSTRING,
+)
+class MaskFormerModel(MaskFormerPretrainedModel):
+    def __init__(self, config: MaskFormerConfig):
+        super().__init__(config)
+        self.pixel_level_module = MaskFormerPixelLevelModule(config)
+        self.transformer_module = MaskFormerTransformerModule(
+            in_features=self.pixel_level_module.encoder.outputs_shapes[-1], config=config
+        )
+
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(MASKFORMER_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_FEAT_EXTRACTOR_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=MaskFormerOutput,
+        config_class=_CONFIG_FOR_DOC,
+        modality="vision",
+    )
+    def forward(
+        self,
+        pixel_values: Tensor,
+        pixel_mask: Optional[Tensor] = None,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = True,
+    ) -> MaskFormerOutput:
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+        batch_size, _, height, width = pixel_values.shape
+
+        if pixel_mask is None:
+            pixel_mask = torch.ones((batch_size, height, width), device=pixel_values.device)
+        pixel_level_module_output: MaskFormerPixelLevelModuleOutput = self.pixel_level_module(
+            pixel_values, output_hidden_states
+        )
+        image_features: Tensor = pixel_level_module_output.encoder_last_hidden_state
+        pixel_embeddings: Tensor = pixel_level_module_output.decoder_last_hidden_state
+
+        transformer_module_output: DetrDecoderOutput = self.transformer_module(image_features, output_hidden_states)
+        queries: Tensor = transformer_module_output.last_hidden_state
+
+        if not return_dict:
+            return (image_features, pixel_embeddings, queries)
+
+        return MaskFormerOutput(
+            encoder_last_hidden_state=image_features,
+            pixel_decoder_last_hidden_state=pixel_embeddings,
+            transformer_decoder_last_hidden_state=queries,
+            encoder_hidden_states=pixel_level_module_output.encoder_hidden_states if output_hidden_states else (),
+            pixel_decoder_hidden_states=pixel_level_module_output.decoder_hidden_states
+            if output_hidden_states
+            else (),
```

Comment:
This formatting is quite unreadable (bad black), so we should probably add parenthesis or have that tensor defined on another line.

Reply:
I've defined the variables above